### PR TITLE
fix: improve Viewer Lite compatibility

### PR DESCRIPTION
**Pull request by ryantrem**
> 🤖 *This PR was created by the create-pr skill.*

## Summary

- Update Viewer dependencies and peer range to published `@babylonjs/lite@1.23.0`.
- Add Lite environment blur/rotation, cubemap HDR skyboxes, rotated shadows, and the new tree-shakeable shadow-only API.
- Fix string hotspot camera orbits and prevent auto-orbit from fighting camera interpolation.
- Add Viewer Lite regression coverage for the compatibility fixes.

## Validation

- Viewer package and source builds
- 34 Viewer Lite Playwright tests
- Lint, formatting, and tree-shaking checks
- Manual WebGPU rendering smoke test

Related: BabylonJS/Babylon-Lite#577 and BabylonJS/Babylon-Lite#572.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by ryantrem** [comment]
Self-review complete using both the Babylon.js instructions and agnostic lenses. Found and fixed one hotspot-camera edge case: omitted camera-orbit components were coerced to zero instead of preserving the current pose. Added regression coverage; targeted Viewer Lite tests pass with REUSE_BROWSER=true. Format, lint, tree-shaking, side-effect sync, and Viewer compile pass. The full unit command currently hits unrelated existing Vitest environment-teardown failures.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/18814/merge

Test environment:
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/index.html#WGZLGJ#4600

Links to test your changes to core in the published versions of the Babylon tools (does not contain changes you made to the tools themselves):

https://playground.babylonjs.com/?snapshot=refs/pull/18814/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/18814/merge
https://gui.babylonjs.com/?snapshot=refs/pull/18814/merge
https://nme.babylonjs.com/?snapshot=refs/pull/18814/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/18814/merge#BCU1XR#0

*If you made changes to the sandbox or playground in this PR, additional comments will be generated soon containing links to the dev versions of those tools.*

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/interactionplaywright/index.html

File: packages/tools/viewer/src/lite/viewer.ts [hunk 1]
```diff
@@ -1081,6 +1091,31 @@ export class Viewer extends ViewerBase implements IViewer {
         light.shadowGenerator = this._shadowGenerator ?? undefined;
     }
 
+    private _rotateShadowLightWithEnvironment(): void {
+        const light = this._shadowLight;
+        const bounds = this._computeModelBounds();
+        if (!light || !bounds) {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (ryantrem) [reply packages/tools/viewer/src/lite/viewer.ts:1099]: [Responded by Copilot on behalf of ryantrem] Added a Viewer Lite regression test that enables normal shadows, applies a dynamic π/2 environment rotation, and verifies the shadow light direction rotates consistently while preserving its Y component. The targeted WebGPU test passes with `REUSE_BROWSER=true`.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/interactionplaywright/index.html

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Interaction tests

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/18814/merge/testResults/interactionplaywright/index.html

**Comment by bjsplat** [comment]
## ⚡ Performance Test Results

🟢 All performance tests passed — no regressions detected.

**Comment by bjsplat** [comment]
## 🟢 Memory Leak Test Results

**4** passed, **0** leaked out of **4** scenarios

🟢 All memory leak tests passed — no leaks detected.

<details>
<summary>Passed Scenarios (4)</summary>

| Scenario | Package |
|----------|---------|
| Core Playground #2FDQT5#1508 | @babylonjs/core |
| Core Playground #T90MQ4#14 | @babylonjs/core |
| Core Playground #8EDB5N#2 | @babylonjs/core |
| Core Playground #LL5BIQ#636 | @babylonjs/core |

</details>

**Review by bghgary — APPROVED** [review]
